grabbit/src/main/groovy/com/twcable/grabbit/server/batch/steps/jcrnodes/JcrNodesProcessor.groovy

```diff
@@ -88,10 +90,51 @@ class JcrNodesProcessor implements ItemProcessor<JcrNode, Node> {
             }
         }
 
-        final List<JcrProperty> properties = jcrNode.properties.toList()
-        Node.Builder nodeBuilder = Node.newBuilder()
+        // It checks if the node parent has required primary type as "nt:hierarchyNode"
+        // and then skip it as the node has already been processed by parent node
+        if(isRequiredHierarchyChildNode(jcrNode)) {
```

Comment:
But if you you follow the design pattern then logic is more suited for `Builder pattern or Factory pattern` rather than `Decorator Pattern`. @jdigger ?